packages/typespec-client-generator-core/src/types.ts

```diff
@@ -190,33 +185,10 @@ function addEncodeInfo(
  * @returns the corresponding sdk built in kind
  */
 function getScalarKind(scalar: Scalar): SdkBuiltInKinds {
-  switch (scalar.name) {
-    case "int8":
-    case "int16":
-    case "int32":
-    case "int64":
-    case "uint8":
-    case "uint16":
-    case "uint32":
-    case "uint64":
-    case "numeric":
-    case "integer":
-    case "safeint":
-    case "decimal128":
-    case "bytes":
-    case "float":
-    case "float32":
-    case "float64":
-    case "boolean":
-    case "string":
-    case "url":
-    case "decimal":
-    case "plainDate":
-    case "plainTime":
-      return scalar.name;
-    default:
-      throw Error(`Unknown scalar kind ${scalar.name}`);
+  if (isSdkBuiltInKind(scalar.name)) {
```

Comment:
Here this is still going to find wong scalars. It should also be checking the namespace

Reply:
If you want what you could do is for every scalar get the fully qualified name and then you create a map of known fully qualified name

```
const knownScalars = {
  "TypeSpec.int32": "...",
 "Azure.Core.eTag": "...",
}
```